lib/oli_web/live/delivery/remix_section.ex

```diff
@@ -848,6 +848,50 @@ defmodule OliWeb.Delivery.RemixSection do
     {:noreply, hide_modal(socket, modal_assigns: nil)}
   end
 
+  def handle_event("show_hide_resource_modal", %{"uuid" => uuid}, socket) do
+    %{hierarchy: hierarchy} = socket.assigns
+
+    node = Hierarchy.find_in_hierarchy(hierarchy, uuid)
+
+    modal_assigns = %{
+      id: "hide_#{uuid}",
+      node: node
+    }
+
+    modal = fn assigns ->
+      ~H"""
+      <HideResourceModal.render {@modal_assigns} />
+      """
+    end
+
+    {:noreply,
+     show_modal(
+       socket,
+       modal,
+       modal_assigns: modal_assigns
+     )}
+  end
+
+  def handle_event("HideResourceModal.toggle", %{"uuid" => uuid}, socket) do
+    %{hierarchy: hierarchy, active: active} = socket.assigns
+
+    hierarchy =
+      Hierarchy.find_and_toggle_hidden(hierarchy, uuid)
+      |> Hierarchy.finalize()
```

Comment:
Finalize is a somewhat expensive operation that updates the numberings, etc. While eventually required before persisting, we likely don't need to call it here since it isn't necessary for rendering this view and will be called upon save